kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java

```diff
@@ -1153,6 +1154,49 @@ public int dbWriteCount(String sql, Object[] fields, String lastField, Connectio
 	}
 
 	/**
+	 * @see org.sakaiproject.db.api.SqlService#dbWriteBatch(Connection, String, Set<Object[]>)
+	 */
+	public boolean dbWriteBatch(Connection callerConnection, String sql, Set<Object[]> fieldsSet)
+	{
+		boolean success = false;
+		PreparedStatement pstmt = null;
+
+		try
+		{
+			pstmt = callerConnection.prepareStatement(sql);
+			for (Object[] fields : fieldsSet)
+			{
+			    prepareStatement(pstmt, fields);
+			    pstmt.addBatch();
+			}
+			pstmt.executeBatch();
+			success = true;
+		}
+		catch (UnsupportedEncodingException e)
+		{
+			LOG.warn("Sql.dbWriteBatch()", e);
+		}
+		catch (SQLException e)
+		{
+			LOG.warn("Sql.dbWriteBatch(): error code: " + e.getErrorCode() + " sql: " + sql + " " + e);
+		}
+		finally
+		{
+			try
```

Comment:
I know this is the existing pattern in this file so changing it makes it stand out but we could use try-with-resources for this instead.

I'm on the fence about this.

Reply:
Yeah I considered but thought maybe we could update the entire file in one swoop instead

Reply:
:+1: for code cleanup!